src/WidgetBase.ts

```diff
@@ -230,22 +228,13 @@ export class WidgetBase<P = WidgetProperties, C extends DNode = DNode> extends E
 	}
 
 	/**
-	 * A render decorator that verifies nodes required in
-	 * 'meta' calls in this render,
+	 * A render decorator that clears the nodehandles map
 	 */
 	@beforeRender()
-	protected verifyRequiredNodes(renderFunc: () => DNode, properties: WidgetProperties, children: DNode[]): () => DNode {
+	protected clearNodeHandlerMap(renderFunc: () => DNode, properties: WidgetProperties, children: DNode[]): () => DNode {
 		return () => {
-			this._requiredNodes.forEach((element, key) => {
-				/* istanbul ignore else: only checking for errors */
-				if (!this._nodeMap.has(key)) {
-					throw new Error(`Required node ${key} not found`);
-				}
-			});
-			this._requiredNodes.clear();
-			const dNodes = renderFunc();
-			this._nodeMap.clear();
-			return dNodes;
+			this._nodeHandler.clear();
```

Comment:
does this still need to be in a `beforeRender` I'd be interested to know if we could remove that.

Reply:
When else would you suggest doing it?

Reply:
could you do before the render call in `__render__`? as all you are doing is creating an extra function that calls the real after clearing the node handler?

Reply:
good call